packages/ctl/ctl.js

```diff
@@ -913,16 +913,13 @@ userCommands.add(new libCommand.Command({
 	handler: async function(args, control) {
 		let data = await fs.readFile(args.file, "utf8");
 		let names = data.split("\n");
-		for (let name in names) {
-			if (name in names) {
-				if (names[name] === "") {
-					continue;
-				}
+		await names.forEach(async (name) => {
```

Comment:
This is wrong. `Array.forEach` returns undefined, which isn't a Promise. You want to do `await Promise.all(names.map(async name => ...` instead

Reply:
This goes for all the places you are using forEach here I see. Hornwitser suggested a for loop. The difference is that promise.all executes in parallel while `for` does sequential. I guess for a long whitelist you'd need to go sequential or batched parallel.

Reply:
Doing it in parallel was my original intent; unfortunately I did mess up the use of forEach. I’ll correct this next commit.